sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/token/generator/impl/EncryptPredicateRightValueTokenGenerator.java

```diff
@@ -79,8 +80,9 @@ private EncryptPredicateRightValueToken generateSQLToken(final EncryptCondition
     
     private EncryptPredicateRightValueToken generateSQLTokenForQueryWithCipherColumn(final EncryptCondition encryptCondition, final List<Object> originalValues) {
         List<Object> encryptedValues = getEncryptedValues(encryptCondition, originalValues);
-        return new EncryptPredicateRightValueToken(encryptCondition.getStartIndex(), encryptCondition.getStopIndex(),  
-                getPositionValues(encryptCondition.getPositionValueMap().keySet(), encryptedValues), encryptCondition.getPositionIndexMap().keySet(), encryptCondition.getOperator());
+        int startIndex = encryptCondition.getOperator().equals(ShardingOperator.IN) ? encryptCondition.getStartIndex() - 1 : encryptCondition.getStartIndex();
```

Comment:
Hi `EncryptPredicateRightValueToken` are created in `generateSQLTokenForQueryWithCipherColumn` and `generateSQLTokenForQueryWithoutCipherColumn`, can you double check whether one more same change is needed?

Reply:
@tristaZero Please review this again lastly. Very sorry to trouble you. Thanks.